sound/soc/msm/msm8x16.c

```diff
@@ -529,6 +507,26 @@ static void msm8x16_ext_spk_delayed_dualmode(struct work_struct *work)
 
     pr_debug("%s: Enable external speaker PAs dualmode.\n", __func__);
 }
+
+static void msm8x16_ext_spk_delayed_disable(struct work_struct *work)
+{
+    int i = 0;
+	
+    /* Close the headset device */
+    gpio_direction_output(EXT_SPK_AMP_HEADSET_GPIO, false);
+    usleep_range(EXT_CLASS_D_EN_DELAY,
+        EXT_CLASS_D_EN_DELAY + EXT_CLASS_D_DELAY_DELTA);
+
+    /* Close external audio PA device */
+    for(i = 0; i < AW8155A_MODE; i++) {
+        gpio_direction_output(EXT_SPK_AMP_GPIO, false);
+	gpio_direction_output(EXT_SPK_AMP_GPIO_1, false);
```

Comment:
I copied with this function https://github.com/boulzordev/android_kernel_lenovo_msm8916/blob/K5_A6020l36_A6020l37/sound/soc/msm/msm8x16.c#L487
Probably he knocked editor.